tests/helpers/execute.ts

```diff
@@ -16,13 +15,23 @@ const createProcess = (processPath: string, args: string[] = [], env: Env = unde
 };
 
 const readStream = async (stream: NodeJS.ReadableStream): Promise<string> => {
-  return new Promise<string>((resolve, reject) => {
-    stream.pipe(
-      concatStream((result) => {
-        resolve(result.toString());
-      }).on('error', reject)
-    );
-  });
+  if (stream.readable) {
+    return new Promise<string>((resolve, reject) => {
+      let str = '';
+      stream.setEncoding('utf-8');
+      stream.on('data', (chunk) => {
+        str += chunk;
+      });
+      stream.on('error', (err) => {
+        reject(err);
+      });
+      stream.on('end', () => {
+        resolve(str);
+      });
+    });
+  } else {
+    return Promise.reject('stream was closed before test was attached');
```

Comment:
fixes comment about returning an error

Reply:
```suggestion
const readStream = async (stream: NodeJS.ReadableStream): Promise<string> => {
  if (!stream.readable) {
    return Promise.reject('stream was closed before test was attached');
  }
  return new Promise<string>((resolve, reject) => {
    let str = '';
    stream.setEncoding('utf-8');
    stream.on('data', (chunk) => {
      str += chunk;
    });
    stream.on('error', reject);
    stream.on('end', () => {
      return resolve(str);
    });
  });
};
```
I think this is much more readable and clean

Reply:
```suggestion
    throw new Error ('The parameter stream is not a readable stream');
```

Reply:
fixed